src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp

```diff
@@ -16,170 +15,147 @@
-                if (SelectTarget(SELECT_TARGET_RANDOM, 0))
-                    DoCastVictim(SPELL_SHADOWBOLT);
-
-                ShadowBolt_Timer = 10000;
-            } else ShadowBolt_Timer -= diff;
+            case EVENT_SHADOWBOLT:
+                if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0))
+                    DoCast(target, SPELL_SHADOWBOLT);
 
-            //Cleave
-            if (Cleave_Timer <= diff)
-            {
+                events.Repeat(10s);
+                break;
+            case EVENT_CLEAVE:
                 DoCastVictim(SPELL_CLEAVE);
-                //13 seconds until we should cast this again
-                Cleave_Timer = 7000 + (rand32() % 10000);
-            } else Cleave_Timer -= diff;
-
-            //MortalStrike
-            if (MortalStrike_Timer <= diff)
-            {
+                events.Repeat(7s, 17s);
+                break;
+            case EVENT_MORTALSTRIKE:
                 DoCastVictim(SPELL_MORTALSTRIKE);
-                MortalStrike_Timer = 10000 + (rand32() % 15000);
-            } else MortalStrike_Timer -= diff;
-
-            //RaiseDead
-            //            if (RaiseDead_Timer <= diff)
-            //          {
-            //      DoCast(me, SPELL_RAISEDEAD);
-            //                RaiseDead_Timer = 45000;
-            //            } else RaiseDead_Timer -= diff;
-
-            //SummonSkeletons
-            if (SummonSkeletons_Timer <= diff)
-            {
-                me->SummonCreature(11197, ADD_POS_1, TEMPSUMMON_TIMED_DESPAWN, 29000);
-                me->SummonCreature(11197, ADD_POS_2, TEMPSUMMON_TIMED_DESPAWN, 29000);
-                me->SummonCreature(11197, ADD_POS_3, TEMPSUMMON_TIMED_DESPAWN, 29000);
-                me->SummonCreature(11197, ADD_POS_4, TEMPSUMMON_TIMED_DESPAWN, 29000);
-                me->SummonCreature(11197, ADD_POS_5, TEMPSUMMON_TIMED_DESPAWN, 29000);
-                me->SummonCreature(11197, ADD_POS_6, TEMPSUMMON_TIMED_DESPAWN, 29000);
-
-                //34 seconds until we should cast this again
-                SummonSkeletons_Timer = 40000;
-            } else SummonSkeletons_Timer -= diff;
-
-            DoMeleeAttackIfReady();
+                events.Repeat(10s, 25s);
+                break;
+            case EVENT_RAISE_DEAD:
+                if (!RaiseDead)
+                {
+                    DoCast(SPELL_RAISE_DEAD);
+                    DoCastSelf(SPELL_RAISE_DEAD1, true);
+                    DoCastSelf(SPELL_RAISE_DEAD2, true);
+                    DoCastSelf(SPELL_RAISE_DEAD3, true);
+                    DoCastSelf(SPELL_RAISE_DEAD4, true);
+                    DoCastSelf(SPELL_RAISE_DEAD5, true);
+                    DoCastSelf(SPELL_RAISE_DEAD6, true);
+                    RaiseDead = true;
+                    Talk(EMOTE_RAISE_DEAD);
+                }
+                else
+                {
+                    DoCast(SPELL_DEATH_PACT);
+                    RaiseDead = false;
```

Comment:
This 2 events has same timers. Baron Rivendare summons these skeletons, after 12 seconds if a skeleton still lives he's getting heal by death pact. Otherwise nothing happens and after 12 seconds he summons again skeletons. I can do that in 2 events, but I guess its easier to make it like that (otherwise I need to schedule event to the death pact in the if part from raise dead event).